File: sdk/c/src/network_services.c
#include "hitechcloud/network_services.h"
#include <stdlib.h>


/* GET /api/service/{id}/ip - List Service IP Addresses */
char *hc_network_services_list_ips(hc_http_client_t *client) {
    return hc_get(client, "/api/service/{id}/ip", NULL);
}

/* GET /api/service/{id}/rdns - Get reverse DNS entries */
char *hc_network_services_get_rdns(hc_http_client_t *client) {
    return hc_get(client, "/api/service/{id}/rdns", NULL);
}

/* POST /api/service/{id}/rdns - Update reverse DNS entries */
char *hc_network_services_update_rdns(hc_http_client_t *client) {
    return hc_post(client, "/api/service/{id}/rdns", NULL);
}

File: sdk/r/R/network_services.R
#' Network Services: IP addresses and reverse DNS for services
#' @param http HttpClient object
#' @return Resource object
#' @export
NetworkServicesResource <- function(http) {
  env <- new.env(parent = emptyenv())
  env$http <- http
  class(env) <- "NetworkServicesResource"
  env
}

#' GET /api/service/{id}/ip - List Service IP Addresses
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
listIps <- function(self, ...) {
  self$http$get("/api/service/{id}/ip", list(...))
}

#' GET /api/service/{id}/rdns - Get reverse DNS entries
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
getRdns <- function(self, ...) {
  self$http$get("/api/service/{id}/rdns", list(...))
}

#' POST /api/service/{id}/rdns - Update reverse DNS entries
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
updateRdns <- function(self, ...) {
  self$http$post("/api/service/{id}/rdns", list(...))
}

